Core/clim-core/presentation-translators.lisp

```diff
@@ -482,53 +482,38 @@ and used to ensure that presentation-translators-caches are up to date.")
-
-(defun map-over-presentations-containing-position (func record x y)
-  "maps recursively over all presentations in record, including record."
-  (map-over-output-records-containing-position
-   #'(lambda (child)
-       (check-type child output-record)
-       (map-over-presentations-containing-position func child x y)
-       #+nil
-       (when (presentationp child)
-         (funcall func child)))
-   record
-   x y)
-  (when (and (presentationp record)
-             (presentation-contains-position record x y))
-    (funcall func record)))
-
-(defun map-applicable-translators (func
-                                   presentation input-context frame window x y
-                                   &key event (modifier-state 0)
-                                     for-menu
-                                     button)
-  (flet ((process-presentation (context context-ptype presentation)
-           (let ((maybe-translators
-                  (find-presentation-translators (presentation-type presentation)
-                                                 context-ptype
-                                                 (frame-command-table frame))))
-             (loop for translator in maybe-translators
-                when (and (or (not for-menu) (eql for-menu (menu translator)))
-                          (test-presentation-translator
-                           translator presentation context-ptype
-                           frame window x y
-                           :event event :modifier-state modifier-state
-                           :for-menu for-menu :button button))
-                do (funcall func translator presentation context)))))
+(defun map-applicable-translators (func presentation input-context frame window x y
+                                   &key event (modifier-state 0) for-menu button)
+  (labels ((process-presentation (context presentation)
+             (let* ((context-ptype (first context))
+                    (maybe-translators
+                     (find-presentation-translators (presentation-type presentation)
+                                                    context-ptype
+                                                    (frame-command-table frame))))
+               (loop for translator in maybe-translators
+                  when (and (or (not for-menu) (eql for-menu (menu translator)))
+                            (test-presentation-translator
+                             translator presentation context-ptype
+                             frame window x y
+                             :event event :modifier-state modifier-state
+                             :for-menu for-menu :button button))
+                  do (funcall func translator presentation context))))
+           (mopscp (context record)
+             "maps recursively over all presentations in record, including record."
+             (map-over-output-records (curry #'mopscp context) record)
```

Comment:
This changes `map-over-output-records-containing-position` to `map-over-output-records`. Are you sure this is an improvement? I'm asking because methods on `map-over-output-records-containing-position` could have a more efficient implementation than linear iteration with position-based filtering.

Reply:
you are right, I've changed the code